components/prize_pool/wikis/trackmania/prize_pool_custom.lua

```diff
@@ -0,0 +1,59 @@
+---
+-- @Liquipedia
+-- wiki=trackmania
+-- page=Module:PrizePool/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+
+local PrizePool = Lua.import('Module:PrizePool', {requireDevIfEnabled = true})
+
+local LpdbInjector = Lua.import('Module:Lpdb/Injector', {requireDevIfEnabled = true})
+local CustomLpdbInjector = Class.new(LpdbInjector)
+
+local CustomPrizePool = {}
+
+local TIER_VALUE = {8, 4, 2}
+local TYPE_MODIFIER = {Online = 0.65}
+
+-- Template entry point
+function CustomPrizePool.run(frame)
+	local args = Arguments.getArgs(frame)
+
+	args.syncPlayers = true
+
+	local prizePool = PrizePool(args):create()
+
+	prizePool:setLpdbInjector(CustomLpdbInjector())
+
+	return prizePool:build()
+end
+
+function CustomLpdbInjector:adjust(lpdbData, placement, opponent)
+	lpdbData.weight = CustomPrizePool.calculateWeight(
+		lpdbData.prizemoney,
+		Variables.varDefault('tournament_liquipediatier'),
+		placement.placeStart,
+		Variables.varDefault('tournament_type')
+	)
+
+	return lpdbData
+end
+
+function CustomPrizePool.calculateWeight(prizeMoney, tier, place, type)
+	if String.isEmpty(tier) then
+		return 0
+	end
+
+	local tierValue = TIER_VALUE[tier] or TIER_VALUE[tonumber(tier) or ''] or 1
```

Comment:
```suggestion
	local tierValue = TIER_VALUE[tonumber(tier)] or 1
```

that should be enough since you only have numeric keys in the `TIER_VALUE` table

Reply:
Just a question because I don't know enough Lua, but since tier can be Weekly for example, can't this crash if you try to do `tonumber("Weekly")` ?

Reply:
`tonumber('Weekly') = nil`
`Table[nil] = nil`

so should be fine

but you can just test it in a sandbox ;)

Reply:
![C89647CD-EA2E-4744-B2AD-B5C81E4D901A](https://user-images.githubusercontent.com/75081997/210090236-99480679-e95c-438b-b8f0-6aa0c2e922d4.jpeg)

Reply:
additionally long term trackmania should switch to standardized tier/tiertype stuff
and in that tiers are numeric anyways

Reply:
Oh yeah didn't think about the sandbox
Thanks for testing, committing the changes

Reply:
fwiw i am about to fix tiers for the standardized infobox to work